### ISSUE TITLE
Asset helper: video

### ISSUE THREAD

**Issue by jodosha** [lotus/lotus#396]
Implement `video` helper.

See [`Lotus::Assets::Helpers`](https://github.com/lotus/assets/blob/master/lib/lotus/assets/helpers.rb) and [`Lotus::Helpers::HtmlHelper`](https://github.com/lotus/helpers/blob/master/lib/lotus/helpers/html_helper.rb) for reference.

### Basic Usage

```erb
<%= video 'movie.mp4' %>
```

```html
<video src="/assets/movie.mp4"></video>
```

### HTML Attributes

It must accept optional HTML attributes.

```erb
<%= video 'movie.mp4', autoplay: true, controls: true %>
```

```html
<video src="/assets/movie.mp4" autoplay controls></video>
```

### Fallback Content

It accepts an optional block for fallback content for browsers not supporting `<video>` tag.

```erb
<%=
  video 'movie.mp4' do
    "Your browser does not support the video tag"
  end
%>
```

```html
<video src="/assets/movie.mp4">
  Your browser does not support the video tag
</video>
```

### Tracks

It should support `<track>` tag.

```erb
<%=
  video 'movie.mp4' do
    track(kind: 'captions', src: asset_path('movie.en.vtt'), srclang: 'en', label: 'English')
    track(kind: 'captions', src: asset_path('movie.it.vtt'), srclang: 'it', label: 'Italiano')
  end
%>
```

```html
<video src="movie.mp4">
  <track kind="captions" src="movie.en.vtt" srclang="en" label="English">
  <track kind="captions" src="movie.it.vtt" srclang="it" label="Italiano">
</video>
```

A few notes:

  * For `<video>` reference, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video
  * In order to avoid concatenation between `track` invocations, use [`Lotus::Helpers::HtmlHelper`](https://github.com/lotus/helpers/blob/master/lib/lotus/helpers/html_helper.rb) internally.
  * The helper used in this example `asset_path`, isn't implemented yet. See https://github.com/lotus/lotus/issues/394

### Sources

It should support `<source>` tag.

```erb
<%=
  video do
    text "Your browser does not support the video tag"
    source src: asset_path('movie.mp4'), type: 'video/mp4'
    source src: asset_path('movie.ogg'), type: 'video/ogg'
  end
%>
```

```html
<video>
  Your browser does not support the video tag
  <source src="/assets/movie.mp4" type="video/mp4">
  <source src="/assets/movie.ogg" type="video/ogg">
</video>
```

A few notes:

  * For `<video>` reference, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video
  * In order to avoid concatenation between `source` invocations, use [`Lotus::Helpers::HtmlHelper`](https://github.com/lotus/helpers/blob/master/lib/lotus/helpers/html_helper.rb) internally.
  * The helper `text` doesn't map any HTML tag, but it's useful to concat pure text. See https://github.com/lotus/helpers/pull/30
  * The helper used in this example `asset_path`, isn't implemented yet. See https://github.com/lotus/lotus/issues/394

### URL Prefix

This should take account of the [`prefix`](https://github.com/lotus/assets/blob/master/lib/lotus/assets/configuration.rb#L45) defined in `Lotus::Assets::Configuration`.

That means, if an application has a prefix: `/admin`, when one this helper is used, the prefix should be used.

```erb
<%= video 'movie.mp4' %>
```

```html
<video src="/assets/admin/movie.mp4"></video>
```

### Digest Mode

If "Digest Mode" is enabled (usually in production), it should generate the file name with the digest suffix (see https://github.com/lotus/assets/blob/master/lib/lotus/assets/configuration.rb#L37).

```erb
<%= video 'song.ogg' %>
```

```html
<video src="/assets/movie-78a21a377046e9ddb3bd9b509c22d4c3.mp4"></video>
```

## Testing Notes

Add an integration test with Lotus::View, in order to make sure that the output isn't escaped twice and produces a valid HTML.

**Comment by leighhalliday** [comment]
I'll give this one a go... my first time contributing to Lotus though so it may take a little bit and I may have some questions :)

**Comment by jodosha** [comment]
@leighhalliday Lovely, thank you! Ping me here on in chat for questions :smile:

**Comment by jodosha** [comment]
Implemented by #407
